ui/src/components/SetupRequired.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react'
+import Ledger from '@daml/ledger';
+import { Form, Button, Loader } from 'semantic-ui-react';
+
+import { Operator } from '@daml.js/da-marketplace/lib/Marketplace/Operator';
+import { httpBaseUrl, deploymentMode, DeploymentMode } from "../config";
+import deployTrigger, { TRIGGER_HASH, MarketplaceTrigger, PublicAutomation, getPublicAutomation } from '../automation';
+import { Tile, logoHeader } from './common/OnboardingTile'
+import { useDablParties } from './common/common';
+
+const SetupRequired  = () => {
+  const [ deploying, setDeploying ] = useState(false);
+  const [ setupError, setSetupError ] = useState(false);
+  const [ adminJwt, setAdminJwt ] = useState('');
+  const { parties } = useDablParties();
+  const [ automations, setAutomations ] = useState<PublicAutomation[] | undefined>([]);
+  const publicParty = useDablParties().parties.publicParty;
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      getPublicAutomation(publicParty).then(autos => {
+        setAutomations(autos);
+        if (!!automations && automations.length > 0) {
+          clearInterval(timer);
+        }
+      });
+    }, 3000);
+    return () => clearInterval(timer);
+  }, [publicParty]);
+
+  const handleSetup = async (event: React.FormEvent) => {
+    setDeploying(true);
+    const ledger = new Ledger({token: adminJwt, httpBaseUrl})
+    try {
+      if (deploymentMode == DeploymentMode.PROD_DABL && TRIGGER_HASH) {
+        deployTrigger(TRIGGER_HASH, MarketplaceTrigger.OperatorTrigger, adminJwt, publicParty);
+      }
+      const args = {
+        operator: parties.userAdminParty,
+        public: parties.publicParty
+      }
+      await ledger.create(Operator, args);
+    } catch(e) {
+      console.log("error exercising setup: " + e);
+      setSetupError(true);
```

Comment:
if the error could be somewhat specific u can make `setupError` a `string | undefined` and set it here, display it below

Reply:
nice 👍